### ISSUE TITLE
Arbitrary lr_scheduler?

### ISSUE THREAD

**Issue by lkhphuc** [williamFalcon/pytorch-lightning#14]
Currently the only learning rate scheduler supported is MultiStepLR, specified through the params of `Trainer()` constructor. 
What do you think about a more flexible approach for lr scheduler, maybe an optional user defined function in `Trainer()` similar to `configure_optimizer`?

**Comment by williamFalcon** [comment]
yeah, that's a good approach.

Two options:
```
trainer = Trainer(...)
trainer.lr_scheduler = SomeScheduler()
```

Or:

```
def learning_rate_scheduler(self):
     return MyScheduler()
```

Option 2 is what you mentioned, and I think probably makes the most sense. It might be that you have multiple models you're running with the same trainer, and each model might use a different learning rate scheduler. Option 2 makes supporting this case easier (otherwise user has to decide which to use based on the model).

What do you think?

**Comment by lkhphuc** [comment]
Or:
```
# In Lightning module
def configure_optimizers(self):
    adam = Adam()
    cosine_anneal = CosineAnneal(adam,..)
    sgd = SGD()
    cyclic = CyclicalLR(sgd)
    return [adam, sgd], [cosine_anneal, cyclic]
  .....
# In Trainer base class
self.optimizers, self.lr_schedulers = model.configure_optimizers()
```
Because each scheduler in pytorch take exactly one optimizer as an arguments, this would make sense to group their definition together.
If no scheduler is needed, the seconds list will be empty.

**Comment by williamFalcon** [comment]
perfect. Mind putting in a PR with these changes? 🙃

**Comment by williamFalcon** [comment]
also need to remove the lr scheduling options from the trainer and update the docs. I can help with any of these if you need

**Comment by williamFalcon** [comment]
Thanks for the feature! fixed a few things, all tests passed and merged!
